************* Module pip._vendor.distlib.locators
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1:0: C0302: Too many lines in module (1300/1000) (too-many-lines)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:64:4: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:64:4: R1710: Either all return statements in a function should return an expression, or none of them should. (inconsistent-return-statements)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:86:0: R0205: Class 'Locator' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:128:16: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:193:8: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:215:12: C0103: Variable name "s1" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:216:12: C0103: Variable name "s2" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:225:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:231:4: R0914: Too many local variables (24/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:247:8: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:275:19: W0703: Catching too general exception Exception (broad-except)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:275:12: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:284:20: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:277:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:275:12: W0612: Unused variable 'e' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:231:4: R0912: Too many branches (20/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:305:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:338:12: C0103: Variable name "md" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:341:12: C0103: Variable name "md" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:365:8: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:388:23: W0703: Catching too general exception Exception (broad-except)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:390:20: W0107: Unnecessary pass statement (unnecessary-pass)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:401:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:402:12: C0103: Variable name "sd" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:351:4: R0912: Too many branches (13/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:149:8: W0201: Attribute '_scheme' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:423:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:436:12: C0103: Variable name "v" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:440:12: E0237: Assigning to attribute 'name' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:441:12: E0237: Assigning to attribute 'version' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:442:12: E0237: Assigning to attribute 'license' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:443:12: E0237: Assigning to attribute 'keywords' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:444:12: E0237: Assigning to attribute 'summary' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:448:16: E0237: Assigning to attribute 'source_url' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:465:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:474:4: R0914: Too many local variables (16/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:522:15: W0703: Catching too general exception Exception (broad-except)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:480:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:481:12: C0103: Variable name "md" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:483:12: E0237: Assigning to attribute 'name' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:484:12: E0237: Assigning to attribute 'version' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:485:12: E0237: Assigning to attribute 'license' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:486:12: E0237: Assigning to attribute 'keywords' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:487:12: E0237: Assigning to attribute 'summary' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:503:16: E0237: Assigning to attribute 'name' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:504:16: E0237: Assigning to attribute 'version' not defined in class slots (assigning-non-slot)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:522:8: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:490:12: W0612: Unused variable 'urls' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:528:0: R0205: Class 'Page' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:551:8: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:572:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:564:8: W0612: Unused variable 'clean' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:528:0: R0903: Too few public methods (1/2) (too-few-public-methods)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:586:0: R0902: Too many instance attributes (14/7) (too-many-instance-attributes)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:610:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:635:12: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:634:12: W0612: Unused variable 'i' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:647:12: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:649:12: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:753:19: W0703: Catching too general exception Exception (broad-except)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:753:12: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:736:8: R1702: Too many nested blocks (8/5) (too-many-nested-blocks)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:762:4: R0914: Too many local variables (18/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:815:23: W0703: Catching too general exception Exception (broad-except)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:799:24: C0103: Variable name "m" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:808:16: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:775:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:811:16: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:815:16: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:762:4: R0912: Too many branches (15/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:762:4: R0915: Too many statements (51/50) (too-many-statements)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:633:8: W0201: Attribute '_threads' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:651:8: W0201: Attribute '_threads' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:656:12: W0201: Attribute 'result' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:657:12: W0201: Attribute 'project_name' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:851:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:857:39: W0613: Unused argument 'parent' (unused-argument)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:868:16: C0103: Variable name "fn" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:870:20: C0103: Variable name "fn" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:867:18: W0612: Unused variable 'dirs' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:887:16: C0103: Variable name "fn" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:889:20: C0103: Variable name "fn" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:886:18: W0612: Unused variable 'dirs' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:927:16: C0103: Variable name "md" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:938:0: W0223: Method 'get_distribution_names' is abstract in class 'Locator' but is not overridden (abstract-method)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:949:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:984:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:987:8: R1725: Consider using Python 3 style super() without arguments (super-with-arguments)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1001:12: C0103: Variable name "d" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1008:20: C0103: Variable name "df" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1010:31: C0103: Variable name "v" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1015:20: C0103: Variable name "dd" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1000:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1000:8: R1702: Too many nested blocks (6/5) (too-many-nested-blocks)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:998:4: R0912: Too many branches (14/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:992:8: W0201: Attribute '_scheme' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1066:0: R0205: Class 'DependencyFinder' inherits from object, can be safely removed from bases in python3 (useless-object-inheritance)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1089:12: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1104:12: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1107:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1173:12: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1186:16: C0103: Variable name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1192:4: R0914: Too many local variables (24/15) (too-many-locals)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1258:20: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1262:16: C0103: Variable name "r" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1275:24: C0103: Variable name "n" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1275:27: C0103: Variable name "v" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1283:20: C0103: Variable name "p" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1192:4: R0912: Too many branches (24/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1192:4: R0915: Too many statements (66/50) (too-many-statements)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1218:8: W0201: Attribute 'provided' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1219:8: W0201: Attribute 'dists' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1220:8: W0201: Attribute 'dists_by_name' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/site-packages/pip/_vendor/distlib/locators.py:1221:8: W0201: Attribute 'reqts' defined outside __init__ (attribute-defined-outside-init)

-----------------------------------
Your code has been rated at 7.75/10

